podcasts/MiniPlayerViewController+Positioning.swift

```diff
@@ -37,11 +37,43 @@ extension MiniPlayerViewController {
         })
     }
 
+    func showMiniPlayerWithDelay() {
+        // only show if something is playing
+        if PlaybackManager.shared.currentEpisode() == nil { return }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            UIView.animate(withDuration: 0.2, animations: { () in
+                self.moveToShownPosition()
+            }, completion: { _ in
+                self.moveToShownPosition() // call this again in case the animation block wasn't called. It's ok to call this twice
+                NotificationCenter.postOnMainThread(notification: Constants.Notifications.miniPlayerDidAppear)
+            })
+        }
+    }
+
     func openFullScreenPlayer(completion: (() -> Void)? = nil) {
         guard PlaybackManager.shared.currentEpisode() != nil else { return }
 
         if playerOpenState == .open || playerOpenState == .animating { return }
 
+        guard !FeatureFlag.newPlayerTransition.enabled else {
+            playerOpenState = .animating
+            aboutToDisplayFullScreenPlayer()
+
+            fullScreenPlayer?.modalPresentationStyle = .overCurrentContext
+
+            presentFromRootController(fullScreenPlayer!, animated: true) {
```

Comment:
I wonder if instead of modifying the MainTabViewController's present method if instead we ensured that the player will always appear by dismissing the presented view if there is one?

However this could result in someone losing their folder their adding, or being kicked out of the upgrade flow. 

An alternative would be to check if there is a presented view and not doing anything. 

Thoughts?

Reply:
> Thoughts?

I think we should follow the system pattern: stack the views by presenting on the top `presentedViewController`. I did that on 31ad6b7cff3e8eea3f15808f28046dcdce6f0f0c

Here's a video of the flow involving purchase and Bookmark:

https://github.com/Automattic/pocket-casts-ios/assets/7040243/cd02a53a-66f6-4456-9726-a7169cc0c21f

Ps.: I ended up changing the sign-out listener too: 2d48e018443055bcbbbdabf4a6b06c9bf5eb0dea